sys/Makefile.dep

```diff
@@ -73,6 +73,10 @@ ifneq (,$(filter base64url,$(USEMODULE)))
   USEMODULE += base64
 endif
 
+ifneq (,$(filter netif,$(USEMODULE)))
+  USEMODULE += fmt
+endif
```

Comment:
This unrelated is still pending as mentioned in https://github.com/RIOT-OS/RIOT/pull/15310#discussion_r516816984, I'll address out of this PR.